### PR TITLE
Addition of Blood Magic Frames

### PR DESCRIPTION
**Pull request by Alastors**
Adds the replacement for the Blood Frame from BM, which is soon to be deleted from BM because of glitches.

Likewise this adds a new frame, The Maddening Frame of Frenzy, 
this has a mutation rate of 10, and a genetic decay of 10 and a durability of 1, 
but consumes LP instead of durability when applicable.

Both use LP, both need a recipe to be decided on a later date, 
both need their lp tweaked, 
the updated LP is currently proposed to be switched to 25k.
They take LP directly from the player network.

Screen_Shot_2023-04-27_at_2 06 04_PM
Screen_Shot_2023-04-27_at_2 06 27_PM
Screen_Shot_2023-04-27_at_2 06 18_PM

### CODE REVIEW

**Comment by Dream-Master** [comment]
Before we merge this please add tooltips to the frames .

**Comment by Alastors** [comment]
> Before we merge this please add tooltips to the frames .

https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/pull/13343

**Comment by Dream-Master** [comment]
@Alastors your pr not build maybe you need to add MB to it ?

**Comment by Alastors** [comment]
MB?

**Comment by Alastors** [comment]
> @Alastors your pr not build maybe you need to add MB to it ?

This is magic bees though? The issue is specific to GitHub and I don't know what's happening here?

**Comment by Alastors** [comment]
The only thing that has worked so far is making BM a hard dependency and that's kind of cringe

**Comment by Dream-Master** [comment]
Fix it we can not merge it before

**Comment by Alastors** [comment]
> Fix it we can not merge it before 

Working on it, it's 3 am where I am rn so I'll get to it in the morning

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:152]: You should probably store them as `Item` to not load the class if there is no dependency, not sure tho as im on the phone

Comment (Alastors) [reply src/main/java/magicbees/main/Config.java]: Nope didn't fix it, any other ideas?

**Comment by Alastors** [comment]
It officially all works now! <3

**Comment by Alastors** [comment]
Screen Shot 2023-04-28 at 1 17 51 PM
Screen Shot 2023-04-28 at 1 18 15 PM
Recipes added

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java:124]: Remove comments

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java:125]: Remove comments

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:213]: Remove this assignment, it is controlled by the config

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:255]: Why are you changing the config variable name? Please revert

**Comment by Alastors** [comment]
Don't merge this yet, I'm going to add an intermediary step that both use so I can gate both to a thaum research

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:494]: Please move this to the BloodHelper as well as the Items uptop here and use it correctly as an IModHelper (register it in the ModHelperManager)

**Review by kuba6000 — CHANGES_REQUESTED** [review]

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java
Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java:124]: Done, will submit as soon as I get this damn research working

File: src/main/java/magicbees/main/Config.java
Comment (Alastors) [reply src/main/java/magicbees/main/Config.java:213]: Can't it's used elsewhere in the mod now

File: src/main/java/magicbees/main/Config.java
Comment (Alastors) [reply src/main/java/magicbees/main/Config.java:255]: Was trying to figure out where the loading issue was

File: src/main/java/magicbees/main/Config.java
Comment (Alastors) [reply src/main/java/magicbees/main/Config.java:494]: Nah that needs to stay where it is, I did switch it out for a boolean though, but the entire mod handles the items like that so.

Comment (Alastors) [reply src/main/java/magicbees/main/Config.java:494]: it has been registered though regardless

**Comment by Alastors** [comment]
I'll fix this crapshow in the morning, I'm gonna have to examine the 9 changes and figure out which broke the camel's back

**Comment by Dream-Master** [comment]
@Alastors i set it draft until it is done

**Comment by Alastors** [comment]
> @Alastors i set it draft until it is done

It should be done as soon as I finish this, I'm just adding back the 9 changes 2 at a time while still making sure it works

**Comment by Alastors** [comment]
> > @Alastors i set it draft until it is done
> 
> It should be done as soon as I finish this, I'm just adding back the 9 changes 2 at a time while still making sure it works

^ so it's easier to revert

**Comment by Alastors** [comment]
Aight, I figured out how to fix it, but Kuba's gonna hate it, the helper being in the helper list is literally what's breaking everything

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:22]: Remove

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:39]: Unused, remove

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:512]: Move to BloodHelper -> getItems

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:258]: Revert

File: src/main/java/magicbees/main/Config.java
Comment (kuba6000) [inline src/main/java/magicbees/main/Config.java:155]: Move to BloodHelper

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:42]: When you move the recipes in the getRecipes, gate this call with isBloodMagicActive check

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:49]: unused?

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:85]: What? Remove

File: src/main/java/magicbees/main/utils/compat/ThaumcraftHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/ThaumcraftHelper.java:446]: Should check for BloodHelper.isActive

File: src/main/java/magicbees/main/utils/compat/ThaumcraftHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/ThaumcraftHelper.java:693]: Should check for BloodHelper.isActive

**Review by kuba6000 — CHANGES_REQUESTED** [review]

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/BloodHelper.java:49]: it was in the others, I don't use it but figured I'd leave it in in case someone wants to add blocks to the compat, but up to you I guess

**Comment by Alastors** [comment]
Thank you for your help Kuba, I do genuinely appreciate the fuck out of it, even if I don't act like it sometimes

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (firenoo) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:32]: What's this for?

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/BloodHelper.java:32]: Leftover from something deleted, it's been removed

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java
Comment (OneEyeMaker) [inline src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java:91]:
```suggestion
        if (par1ItemStack.hasTagCompound()) {
```

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java
Comment (OneEyeMaker) [inline src/main/java/magicbees/main/utils/compat/bloodmagic/ItemFrenziedFrame.java:123]: Same issue.

**Review by OneEyeMaker — CHANGES_REQUESTED** [review]

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java
Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: That's intended

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java
Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java:124]: no, I did it as intended

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java
Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java:21]: This was kuba's suggestion but sure, whatever

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodFrame.java:21]: I'm going to remind you though, this can't be stacked in the first place

File: src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java
Comment (OneEyeMaker) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: @Alastors Imagine situation (with current implementation of these frames). I join server with many players and craft this frame. And at the moment I bind this frame to my network ***all*** such frames in all hives on server begin to drain LP from my network - and not from network of owner on tooltip.

Is ***that*** intended?

Comment (OneEyeMaker) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: Because my thought is that frame should drain LP from network of it's owner (like all BM tools do). And not from network of person who crafted and bound new frame.

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: That's not what would happen

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: And I'll prove it, just a moment

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: Screen Shot 2023-04-30 at 10 27 50 AM
Screen Shot 2023-04-30 at 10 27 58 AM
Screen Shot 2023-04-30 at 10 28 25 AM
Screen Shot 2023-04-30 at 10 28 34 AM
Screen Shot 2023-04-30 at 10 28 42 AM
@OneEyeMaker you were wrong, here's proof of such, if you were right this would've taken 2k out of my LP network, but no, it did not, it took 1k and the other frame dissolved like intended

Comment (Alastors) [reply src/main/java/magicbees/main/utils/compat/bloodmagic/ItemBloodBaseFrame.java:124]: I can even film this working in action if you want me to

**Comment by Alastors** [comment]
I'll put this out here and be done with it, I have no ability to test this on multiplayer currently, someone else would need to do that as such.

**Review by OneEyeMaker — APPROVED** [review]

File: src/main/java/magicbees/main/utils/compat/BloodHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BloodHelper.java:40]: Please add config check
```suggestion
        if (Loader.isModLoaded(Name) && Config.bloodMagicActive) {
```

File: src/main/java/magicbees/main/utils/compat/BaublesHelper.java
Comment (kuba6000) [inline src/main/java/magicbees/main/utils/compat/BaublesHelper.java:22]: Add config

**Review by kuba6000 — CHANGES_REQUESTED** [review]

**Pull request merged**
